Wabbajack.Lib/AInstaller.cs

```diff
@@ -233,6 +234,22 @@ public async Task DownloadArchives()
 
             Info("Getting Nexus API Key, if a browser appears, please accept");
 
+            var client = await NexusApiClient.Get();
+            bool premium = await client.IsPremium();
+
+            //If we aren't premium, fall back to manual downloading method by converting states.
+            if (!premium)
+            {
+                Utils.Log($"Nexus user {await client.Username()} is not a premium account. Falling back to manual downloads.");
+                foreach (var a in missing.Where(a => a.State.GetType() == typeof(NexusDownloader.State)))
```

Comment:
Use the `Do` Extension from `Wabbajack.Common.Utils`:
```cSharp
missing.Where(a => a.State.GetType() == typeof(NexusDownloader.State)).Do(f => ....)
```

Reply:
Haha, I've thought a few times of commenting that we should swap `.Do`s for foreach loops. 8P
I think it's pretty heavily personal preference.  I do think a forloop is technically faster as you dont have any lambda variable captures, but that's pretty minor reason to care for most situations.  I'd say it's not much of a problem either way